pkg/streamingpromql/engine_test.go

```diff
@@ -2921,7 +2987,7 @@ func runMixedMetricsTests(t *testing.T, expressions []string, pointsPerSeries in
 	// - Look backs
 
 	opts := NewTestEngineOpts()
-	mimirEngine, err := NewEngine(opts, NewStaticQueryLimitsProvider(0), stats.NewQueryMetrics(nil), log.NewNopLogger())
+	mimirEngine, err := NewEngine(opts, NewStaticQueryLimitsProvider(0), stats.NewQueryMetrics(nil), nil, log.NewNopLogger())
```

Comment:
I'd be curious to run the gauntlet through w/wout the planner. Maybe not permanently, but just as an early validation?

Reply:
Done: https://github.com/grafana/mimir/pull/11124/commits/7e81b8f1b96ff89ed0b094659a30589e478403d9